File: src/Other/MicroPython/command.py
class Account:
    def __init__(self, balance):
        self.balance = balance

    def deposit(self, amount):
        self.balance += amount

    def withdraw(self, amount):
        self.balance -= amount


class Command:
    def __init__(self, action, amount):
        self.action = action
        self.amount = amount

    def execute(self):
        self.action(self.amount)


account = Account(100)
queue = [Command(account.deposit, 50), Command(account.withdraw, 20)]
for command in queue:
    command.execute()
assert account.balance == 130
print("balance={};commands={}".format(account.balance, len(queue)))

File: src/DataScience/Julia/patterns/command.jl
abstract type Command end

struct Deposit <: Command
    amount::Int
end

struct Withdraw <: Command
    amount::Int
end

execute(balance::Int, command::Deposit) = balance + command.amount
execute(balance::Int, command::Withdraw) = balance - command.amount

queue = Command[Deposit(50), Withdraw(20)]
balance = foldl(execute, queue; init = 100)
@assert balance == 130
@assert length(queue) == 2
println("balance=$(balance);commands=$(length(queue))")
